### PR TITLE
Agents pagination

### PR DESCRIPTION
**Pull request by nicolasparada**
# Summary of this proposal

Adding tests for the new paginated agents.
Adding the spec for the new paginated agents endpoint.
Update readme TS client codegen command to include a client.

## Notes for the reviewer

The generated TS client doesn't expose the headers in the resolved responses. See https://github.com/ferdikoomen/openapi-typescript-codegen/blob/b3530b14fc82ba285ff7940c750f50f37dd32bdb/test/__snapshots__/index.spec.ts.snap#L524

Related issue https://github.com/ferdikoomen/openapi-typescript-codegen/issues/829

We might have to switch the code generator, or fork it, or take a look at the `--request` option.

## Issue(s) number(s)

https://github.com/calyptia/api/issues/50

## Checklist for submitter

- [ ] My PR has a related issue/bug number.
- [x] My PR provides tests
  - [x] Integration tests are added/passes
  - [ ] Unit tests are added/passes
  - [ ] End-to-end tests added/passes
  - [ ] Static code check added/passes
  - [ ] Linting on documentation added/passes
  - [ ] Does not affect code coverage stats
- [ ] My PR requires updating dependencies
- [ ] My PR has the documentation changes required.

## Checklist for reviewer

- [ ] The proposal fixes a bug/issue or implements a new feature that is well described.
- [ ] The proposal has sufficient test cases that covers the changes.
  - [ ] If changes an API, it does not break backwards compatibility (-1 major version).
- [ ] If integration is required, the proposal has integration tests.
- [ ] The proposal does not break coverage stats
- [ ] The proposal has the required documentation changes.

## Backport

- [ ] Backport to the latest stable release.

### CODE REVIEW

File: spec/open-api.yml [hunk 1]
```diff
@@ -1874,6 +1879,17 @@ paths:
...
+            Link:
+              schema:
+                type: string
+                example: </v1/projects/foo/agents?before=bar>; rel="next"
+              description: RFC5988 with `rel="next"`.
+          links:
+            NextAgentsPage:
+              operationId: agents
+              parameters:
+                before: "$response.header.Link#rel=next"
```
Comment (nicolasparada) [inline spec/open-api.yml:1892]: The part after the hash `#` is not totally correct. I didn't found any tool that support it. But it is also not incorrect. I left it there for documentation puposes.

**Comment by nicolasparada** [comment]
CC @gagoar @mlacosta

**Comment by niedbalski** [comment]
check the linter details @nicolasparada

File: spec/open-api.yml (hunk 1 above)
Comment (nicolasparada) [reply spec/open-api.yml:1892]: Maybe we could remove the `links` field part. For example GitHub's API uses `Link` header bases pagination https://github.com/github/rest-api-description/blob/main/descriptions/api.github.com/api.github.com.yaml but it doesn't document the links field.

**Comment by nicolasparada** [comment]
> check the linter details @nicolasparada

Strange... `Has()` has been there since 1.17 https://pkg.go.dev/net/url#Values.Has

File: client/client.go [hunk 2]
```diff
@@ -130,3 +148,40 @@ func (c *Client) do(ctx context.Context, method, path string, v, dest interface{
 
 	return nil
 }
+
+type opts struct {
+	cursor **string
```
[the remark this answers is not in the document]
Comment (nicolasparada) [reply client/client.go:153]: Mostly to check on not nil.
Take a look at were it called.
So i don't have to initialize a pointer.

[the remark this answers is not in the document]
Comment (nicolasparada) [reply client/client.go:153]: But I could have used a simple pointer too.

**Comment by nicolasparada** [comment]
Fixed CI by setting go before running golangci-lint.

> v3.0.0+ requires explicit setup-go installation step prior to using this action: uses: actions/setup-go@v2

File: client/client.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (niedbalski) [reply client/client.go:156]: > But I could have used a simple pointer too.

I'd stick to this.

[the remark this answers is not in the document]
Comment (nicolasparada) [reply client/client.go:156]: I was trying now with simple pointer but couldn't make it work.
In my head it has more sense in this way, but cannot explain clearly why 😬

**Pull request merged**